File: benchmarks/sieve/sieve.lua
local clock = os.clock

local t0 = clock()

local N = tonumber(arg[1])
local x = {}

local i = 0
while i < N do
    x[i] = true
    i = i + 1
end

i = 2
while i * i < N do
    if x[i] then
        local j = 2 * i
        while j < N do
            x[j] = false
            j = j + i
        end
    end
    i = i + 1
end

print("Primes until: " .. N)
i = 0
while i < N do
    if x[i] then
        print(i)
    end
    i = i + 1
end

local t1 = clock()
print(string.format("Execution took (ms) " .. (t1 - t0) * 1000))

File: benchmarks/sieve/sieve.py
import time
import sys

t0 = time.perf_counter()

N = int(sys.argv[1])
x = []

i = 0
while i < N:
    x.append(True)
    i += 1

i = 2
while i * i < N:
    if not x[i]:
        i += 1
        continue
    j = 2 * i
    while j < N:
        x[j] = False
        j += i
    i += 1

print("Primes until: " + str(N))
i = 0
while i < N:
    if x[i]:
        print(i)
    i += 1

t1 = time.perf_counter()
print("Execution took (ms) " + str((t1 - t0) * 1000))
